demisto_sdk/commands/common/docker/tests/dockerhub_client_test.py

```diff
@@ -0,0 +1,246 @@
+    Given:
+        - existing token from the cache that is not expired
+
+    When:
+        - running get_token method
+
+    Then:
+        - ensure that the token is extracted properly only from the cache without api-request
+    """
+    dockerhub_client._docker_hub_auth_tokens = {
+        "test:pull": {
+            "token": "1234",
+            "issued_at": (datetime.now() - timedelta(minutes=8)).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ"
+            ),
+            "expires_in_seconds": 300,
+        }
+    }
+    requests_mock.get("https://auth.docker.io/token")
+    assert dockerhub_client.get_token(repo="test") == "1234"
+    assert not requests_mock.called
+
+
+def test_get_token_with_existing_expired_token(
+    requests_mock, dockerhub_client: DockerHubClient
+):
+    """
+    Given:
+        - existing token from the cache that is expired
+
+    When:
+        - running get_token method
+
+    Then:
+        - ensure that the token is extracted from the api-request because token has expired
+        - ensure the api is called
+        - ensure that cache gets updated with the newly created token
+    """
+    dockerhub_client._docker_hub_auth_tokens = {
+        "test:pull": {
+            "token": "token_from_cache",
+            "issued_at": (datetime.now() - timedelta(minutes=1)).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ"
+            ),
+            "expires_in_seconds": 300,
+        }
+    }
+    requests_mock.get(
+        "https://auth.docker.io/token",
+        json={"token": "token_from_api", "issued_at": "1234", "expires_in": 300},
+    )
+    assert dockerhub_client.get_token(repo="test") == "token_from_api"
+    assert requests_mock.called
+    assert dockerhub_client._docker_hub_auth_tokens["test:pull"] == {
+        "token": "token_from_api",
+        "issued_at": "1234",
+        "expires_in_seconds": 300,
+    }
+
+
+@pytest.mark.parametrize(
+    "tags, expected_highest_tag",
+    [
+        (
+            [
+                "3.10.13.78960",
+                "3.10.13.81631",
+                "3.10.12.68300",
+                "3.10.11.61265",
+                "3.8.3.9324",
+                "3.9.8.24399",
+            ],
+            "3.10.13.81631",
+        ),
+        (["1.0.0.81877", "1.0.0.78900", "1.0.0.72295"], "1.0.0.81877"),
+        (["1.0.0.81877", "1.0.0.78900", "2.0.0.72295"], "2.0.0.72295"),
+        (
+            [
+                "invalid_value",
+                "7.4.0.80528",
+                "1.5.0.80528",
+                "2.5.0.80528",
+                "2.5.0.80529",
+            ],
+            "7.4.0.80528",
+        ),
+    ],
+)
+def test_get_latest_docker_image_tag(
+    requests_mock,
+    dockerhub_client: DockerHubClient,
+    tags: List[str],
+    expected_highest_tag: str,
+):
+    """
+    Given:
+        - lists of tags
+
+    When:
+        - running get_latest_docker_image_tag method
+
+    Then:
+        - ensure that the latest tag is returned always
+    """
```

Comment:
Maybe add more info about the test cases here?

Reply:
its pretty straight forward, just take the largest version tag, there is no any special case here except maybe handling invalid tags / values.